test/test_nn.py

```diff
@@ -1770,6 +1770,24 @@ def test_Conv2d_inconsistent_types_on_GPU_with_cudnn(self):
         # but it should work with the same type
         nn.functional.conv2d(inputs.float(), weights.float(), bias.float())
 
+    @unittest.skipIf(not TEST_CUDA, 'CUDA not available')
+    @unittest.skipIf(not TEST_CUDNN, 'CUDNN not available')
+    def test_Conv2d_deterministic_on_GPU_with_cudnn(self):
+        inputs = Variable(torch.randn(4, 1, 7, 7).float().cuda())
+        weights = Variable(torch.randn(1, 1, 3, 3).double().cuda())
+        bias = Variable(torch.randn(1).double().cuda())
```

Comment:
why are your weights of different type than input? how could that even pass a contbuild?

Reply:
I guess I took it from the previous tests and line 1787-1788 typecasts them before passing so contbuild should work.